lib/rules/no-curly-component-invocation.js

```diff
@@ -47,14 +49,47 @@ const BUILT_INS = new Set([
 
 const ALWAYS_CURLY = new Set(['yield']);
 
+function toAngleAttrValue(attr) {
+  const node = attr.value;
+  if (node.type === 'SubExpression') {
+    if (node.path.original === 'concat') {
+      // eslint-disable-next-line unicorn/prefer-spread
+      return b.concat(
+        node.params.map((p) => {
+          if (p.type === 'PathExpression') {
+            return b.mustache(p);
+          } else if (p.type === 'StringLiteral') {
```

Comment:
This should probably handle other literal types. For example:

```hbs
{{#foo-bar 123}}{{/foo-bar}}
{{#foo-bar true}}{{/foo-bar}}
{{#foo-bar false}}{{/foo-bar}}
{{#foo-bar undefined}}{{/foo-bar}}
{{#foo-bar null}}{{/foo-bar}}
```

Reply:
related - https://github.com/ember-template-lint/ember-template-lint/pull/1594#discussion_r579617691

Reply:
I realized that my comment was slightly off. This conditional is specifically for `(concat` helper. From the context of `concat` I think you are right, all types would be coerced to string anyways.